datadog/resource_datadog_logs_archive.go

```diff
@@ -0,0 +1,355 @@
+	authV2 := providerConf.AuthV2
+	ddArchive, httpresp, err := datadogClientV2.LogsArchivesApi.GetLogsArchive(authV2, d.Id()).Execute()
+	if err != nil {
+		// API returns 404 when the specific archive id doesn't exist.
+		if httpresp != nil && httpresp.StatusCode == 404 {
+			return false, nil
+		}
+		return false, translateClientError(err, "error checking if logs archive exists")
+	}
+	return ddArchive.HasData(), nil
+}
+
+//Model to map
+func buildDestination(archiveDestination datadogV2.NullableLogsArchiveDestination) (string, map[string]interface{}, error) {
+	emptyDestination := map[string]interface{}{}
+	if archiveDestination.IsSet() {
+		destination := archiveDestination.Get().GetActualInstance()
+		switch d := destination.(type) {
+		case *datadogV2.LogsArchiveDestinationAzure:
+			return "azure", buildAzureMap(*d), nil
+		case *datadogV2.LogsArchiveDestinationGCS:
+			return "gcs", buildGCSMap(*d), nil
+		case *datadogV2.LogsArchiveDestinationS3:
+			return "s3", buildS3Map(*d), nil
+		}
+	}
+	return "", emptyDestination, fmt.Errorf("Destination should be not null.")
+}
+
+func buildAzureMap(destination datadogV2.LogsArchiveDestinationAzure) map[string]interface{} {
+	result := make(map[string]interface{})
+	result["client_id"] = destination.Integration.ClientId
+	result["tenant_id"] = destination.Integration.TenantId
+	result["container"] = destination.Container
+	result["storage_account"] = destination.StorageAccount
+	result["path"] = destination.Path
+	return result
+}
+
+func buildGCSMap(destination datadogV2.LogsArchiveDestinationGCS) map[string]interface{} {
+	result := make(map[string]interface{})
+	result["client_email"] = destination.Integration.ClientEmail
+	result["project_id"] = destination.Integration.ProjectId
+	result["bucket"] = destination.Bucket
+	result["path"] = destination.Path
+	return result
+}
+
+func buildS3Map(destination datadogV2.LogsArchiveDestinationS3) map[string]interface{} {
+	result := make(map[string]interface{})
+	result["account_id"] = destination.Integration.AccountId
+	result["role_name"] = destination.Integration.RoleName
+	result["bucket"] = destination.Bucket
+	result["path"] = destination.Path
+	return result
+}
+
+//Map to model
+func buildDatadogArchiveCreateReq(d *schema.ResourceData) (datadogV2.LogsArchiveCreateRequest, error) {
+	archive := datadogV2.NewLogsArchiveCreateRequest()
+	destination, err := buildCreateReqDestination(d)
+	if err != nil {
+		return *archive, err
+	}
+	attributes := datadogV2.NewLogsArchiveCreateRequestAttributes(
+		destination,
+		d.Get("name").(string),
+		d.Get("query").(string),
+	)
+	definition := datadogV2.NewLogsArchiveCreateRequestDefinitionWithDefaults()
+	definition.SetAttributes(*attributes)
+	archive.SetData(*definition)
+	return *archive, nil
+}
+
+func buildCreateReqDestination(d *schema.ResourceData) (datadogV2.LogsArchiveCreateRequestDestination, error) {
+	emptyDestination := datadogV2.LogsArchiveCreateRequestDestination{}
```

Comment:
Would it be possible to use `datadogV2.New...` methods instead (here and similar places)